src/Matroids/properties.jl

```diff
@@ -0,0 +1,302 @@
+fundamental_cocircuit(M::Matroid, basis::AbstractVector, elem::Union{Int,Char,String}) = fundamental_circuit(dual_matroid(M),basis,elem)
+
+@doc Markdown.doc"""
+    independet_sets(M::matroid)
+
+Return the list of independent sets of the matroid `M`.
+
+# Example
+```jldoctest
+julia> 
+
+```
+"""
+function spanning_sets(M::Matroid)
+	pm_bases = Vector{Set{Int}}(M.pm_matroid.BASES)
+	n = size(M.groundset)[1]
+	gs = M.groundset
+	sets = Vector()
+	for k in rank(M):n
+		for set in Oscar.Hecke.subsets(Vector(0:n-1),k)
+			for B in pm_bases
+				if issubset(B,set)
+					push!(sets, [Vector([gs[i+1] for i in set])])
+					break
+				end
+			end
+		end
+	end
+	return sets
+end
+
+function independent_sets(M::Matroid)
+	pm_bases = Vector{Set{Int}}(M.pm_matroid.BASES)
+	n = size(M.groundset)[1]
+	gs = M.groundset
+	sets = Vector(undef,1)
+	for k in 1:rank(M)
+		for set in Oscar.Hecke.subsets(Vector(0:n-1),k)
+			for B in pm_bases
+				if issubset(set,B)
+					push!( sets, [Vector([gs[i+1] for i in set])])
+					break
+				end
+			end
+		end
+	end
+	sets[1]=[]
+	return sets
+end
+
+cobases(M::Matroid) = bases( dual_matroid(M) )
+
+cocircuits(M::Matroid) = circuits( dual_matroid(M) )
+
+cohyperplanes(M::Matroid) = hyperplanes( dual_matroid(M) )
+
+corank(M::Matroid, set::Vector) = size(set)[1]-rank(M, set) + rank(M, setdiff(M.groundset,set))
+
+function is_clutter(sets::Union{AbstractVector{<:AbstractVector}, AbstractVector{<:AbstractSet}})
+	for A in sets
+		for B in sets
+			if issubset(A,B) ⊻ issubset(B,A)
+				return false
+			end
+		end
+	end
+	return true
+end
+
+function get_blocker(clutter::Union{AbstractVector{<:AbstractVector}, AbstractVector{<:AbstractSet}}, groundset::Vector)
+	#TODO
+end
+
+is_regular(M::Matroid) = M.pm_matroid.REGULAR
+
+is_graphic(M::Matroid) = true #TODO
+
+is_cographic(M::Matroid) = is_graphic(dual_matroid(M))
+
+is_binary(M::Matroid) = M.pm_matroid.BINARY
+
+is_ternary(M::Matroid) = M.pm_matroid.TERNARY
+
+n_connected_components(M::Matroid) = size(M.pm_matroid.CONNECTED_COMPONENTS)[1]
+
+connected_components(M::Matroid) = [[M.groundset[i+1] for i in comp] for comp in M.pm_matroid.CONNECTED_COMPONENTS]
+is_connected(M::Matroid) = M.pm_matroid.CONNECTED
+
+loops(M::Matroid) = [M.groundset[i+1] for i in M.pm_matroid.LOOPS]
+
+coloops(M::Matroid) = [M.groundset[i+1] for i in M.pm_matroid.DUAL.LOOPS]
+
+is_loopless(M::Matroid) = length(M.pm_matroid.LOOPS)==0 ? true : false
+
+is_coloopless(M::Matroid) = length(M.pm_matroid.DUAL.LOOPS)==0 ? true : false
+
+function direct_sum_components(M::Matroid)
+	res = Vector{Matroid}()
+	for set in connected_components(M)
+		res = [res; restriction(M,set)]
+	end
+	return res
+end
+
+connectivity_function(M::Matroid, set::Union{AbstractVector, Set}) = rank(M,set) + rank(M,setdiff( Set(M.groundset), set)) - rank(M)
+
+is_vertical_k_separation(M::Matroid,k::Int, set::Union{AbstractVector,Set}) = k<=rank(M,set) && k<=rank(M,setdiff( Set(M.groundset), set)) && k> connectivity_function(M,set)
```

Comment:
This is not a hard rule, but: I find very long lines (say... > 100 chars or so, but again, this is not hard) to be difficult to read. In this case the line is longer than what I can see at once in the GitHub code review. So for these things, I'd recommend just writing a full function, say
```suggestion
function is_vertical_k_separation(M::Matroid, k::Int, set::Union{AbstractVector,Set})
  return k <= rank(M,set) && k <= rank(M, setdiff(Set(M.groundset), set)) && k >  connectivity_function(M, set)
end
```